options.go

```diff
@@ -76,7 +76,7 @@ type Options struct {
 	// Default is 3 seconds.
 	ReadTimeout time.Duration
 	// Timeout for socket writes. If reached, commands will fail
-	// with a timeout instead of blocking.
+	// with a timeout instead of blocking. Use value -1 for no timeout and 0 for default.
```

Comment:
`-1 for no timeout` is meant to disable the timeout, but it looks like you always call `SetReadDeadline` no matter what. Perhaps the comment must be updated?

Reply:
This sentience is copied from the `ReadTimeout` description above.
I think `no timeout` == `disable the timeout` == `block until read/write succeeds`, do I understand this correctly?

`-1` will be adjusted to `0` here: https://github.com/go-redis/redis/blob/084c0c8914d4d84918a81b867b61d94b6c343ecd/options.go#L152-L157
then in `SetReadDeadline` or `SetWriteDeadline`, the timeout is diabled, and the read/write operation will block until it succeeds.

Reply:
That is correct, but it looks like we should also support `-2` to disable `Set*Deadline` calls completely.